data-prep-pipeline/scripts/subtype_tree/create_context_matrix.py

```diff
@@ -0,0 +1,244 @@
+        #this context
+        st_models = get_context_models(
+            subtype_node=st, 
+            subtype_tree=subtype_tree, 
+            genetic_subtypes=genetic_subtypes,
+            model_tree=model_tree)
+        
+        #determine which column contains the code
+        if st.NodeSource in ['Oncotree', 'Depmap']: 
+            ctx_code = st.DepmapModelType
+
+        else:
+            ctx_code = st.MolecularSubtypeCode
+
+        #create a one-hot encoded series object for this context
+        st_ctx = pd.Series(
+                    index=model_tree.ModelID,
+                    name=ctx_code,
+                    data=0
+                )
+        st_ctx.loc[st_models] = 1
+
+        #save the series to the list of columns
+        ctx_cols.append(st_ctx)
+
+    context_matrix = pd.DataFrame(ctx_cols).T
+
+    return context_matrix
+
+def sanity_check_results(subtype_tree, context_matrix):
+    '''
+    A function to make sure that the final result of the context matrix does not
+    break any assumptions/rules that are used by the portal
+
+    Inputs:
+        - subtype_tree (pandas df): The subtype tree as created by create_subtype_tree.py
+        - context_matrix (pandas df): The final result of the context matrix
+
+    Outputs: None (function will raise an error if any assert fails)
+    '''
+
+    #assert that all column names are unique
+    assert(len(context_matrix.columns) == len(context_matrix.columns.unique()))
+
+    #assert that all nodes in the subtype tree have a corresponding column
+    cm_nodes = set(context_matrix.columns)
+    st_nodes = set(subtype_tree.DepmapModelType.dropna())\
+                .union(set(subtype_tree.MolecularSubtypeCode.dropna()))
+
+    assert(set.intersection(cm_nodes, st_nodes) == set.union(cm_nodes, st_nodes))
+
+def create_context_matrix(
+        model_taiga_id,
+        molecular_subtypes_taiga_id,
+        subtype_tree_path
+    ):
+    model_tree, subtype_tree, genetic_subtypes = load_data(
+        model_taiga_id,
+        molecular_subtypes_taiga_id,
+        subtype_tree_path
+    )
+
+    context_matrix = construct_matrix(
+        model_tree,
+        subtype_tree,
+        genetic_subtypes
+    )
+
+    sanity_check_results(subtype_tree, context_matrix)
+
+    return context_matrix
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="Create ContextMatrix"
+    )
+    parser.add_argument("model", help="Taiga ID of model table")
+    parser.add_argument("molecular_subtypes", help="Taiga ID of Omics Inferred Molecular Subtypes")
+    parser.add_argument("subtype_tree", help="Filepath for the SubtypeTree")
+    parser.add_argument("output", help="filepath to write the output")
+    args = parser.parse_args()
+
+    context_matrix = create_context_matrix(
+        args.model,
+        args.molecular_subtypes,
+        args.subtype_tree
+    )
+
+    if context_matrix is not None:
```

Comment:
Is there ever a case where the context matrix being None isn't a gigantic problem? If None is not an acceptable result, should we just let this blow up here so we're aware sooner?